repos/spack_repo/builtin/packages/palace/package.py

```diff
@@ -3,92 +3,107 @@
     variant(
         "superlu-dist", default=True, description="Build with SuperLU_DIST sparse direct solver"
     )
     variant("strumpack", default=False, description="Build with STRUMPACK sparse direct solver")
+    variant(
+        "sundials",
+        default=True,
+        description="Build with SUNDIALS differential/algebraic equations solver",
+        when="@0.14:",
+    )
     variant("mumps", default=False, description="Build with MUMPS sparse direct solver")
     variant("slepc", default=True, description="Build with SLEPc eigenvalue solver")
     variant("arpack", default=False, description="Build with ARPACK eigenvalue solver")
-    variant("libxsmm", default=True, description="Build with LIBXSMM backend for libCEED")
-    variant("magma", default=True, description="Build with MAGMA backend for libCEED")
+    variant("libxsmm", default=True, description="Build with libxsmm backend for libCEED")
     variant(
         "gslib",
         default=True,
         description="Build with GSLIB library for high-order field interpolation",
     )
 
-    # Dependencies
+    # -- Patches for Spack Build System --
+    # Fix API mismatch between libxsmm@main and internal libceed build
+    patch("palace-0.12.0.patch", when="@0.12")
+
+    # -- Core Dependencies --
+    # NOTE: We can't depend on git tagged versions here
+    #       https://github.com/spack/spack/issues/50171
+    #       Instead, version in environment / spec
     depends_on("c", type="build")
     depends_on("cxx", type="build")
-
     depends_on("cmake@3.21:", type="build")
     depends_on("pkgconfig", type="build")
     depends_on("mpi")
     depends_on("zlib-api")
     depends_on("nlohmann-json")
-    depends_on("fmt")
+    depends_on("fmt+shared", when="+shared")
+    depends_on("fmt~shared", when="~shared")
     depends_on("eigen")
 
-    depends_on("metis@5:")
-    depends_on("metis+shared", when="+shared")
-    depends_on("metis~shared", when="~shared")
-    depends_on("metis+int64", when="+int64")
-    depends_on("metis~int64", when="~int64")
+    # -- Sparse Direct Solvers --
+    conflicts("~superlu-dist~strumpack~mumps", msg="Need at least one sparse direct solver")
 
-    depends_on("hypre~complex")
-    depends_on("hypre+shared", when="+shared")
-    depends_on("hypre~shared", when="~shared")
-    depends_on("hypre+mixedint", when="+int64")
-    depends_on("hypre~mixedint", when="~int64")
-    depends_on("hypre+openmp", when="+openmp")
-    depends_on("hypre~openmp", when="~openmp")
+    # -- MUMPS --
+    conflicts("^mumps+int64", msg="Palace requires MUMPS without 64 bit integers")
+    with when("+mumps"):
+        depends_on("mumps+metis+parmetis")
+        depends_on("mumps+shared", when="+shared")
+        depends_on("mumps~shared", when="~shared")
+        depends_on("mumps+openmp", when="+openmp")
+        depends_on("mumps~openmp", when="~openmp")
 
+    # -- SuperLU-Dist --
```

Comment:
Can we leave out these kind of comments (also for other dependencies)?
```suggestion
```

Reply:
I did this in a single commit in case @hughcars wants to revert.